File: src/main/java/com/ggardet/patient/spi/PatientService.java
package com.ggardet.patient.spi;

import com.ggardet.patient.spi.dto.PatientDTO;

import java.util.List;
import java.util.UUID;

public interface PatientService {
    PatientDTO create(final PatientDTO patientDTO);
    PatientDTO findById(final UUID id);
    List<PatientDTO> findAll();
    List<PatientDTO> findByHospitalId(final UUID hospitalId);
    List<PatientDTO> findByServiceId(final UUID serviceId);
    PatientDTO findBySocialSecurityNumber(final String socialSecurityNumber);
    List<PatientDTO> searchByName(final String name);
    PatientDTO update(final UUID id, final PatientDTO patientDTO);
    void deleteById(final UUID id);
}

File: src/main/kotlin/com/ggardet/patient/spi/PatientService.kt
package com.ggardet.patient.spi

import com.ggardet.patient.spi.dto.PatientDTO
import java.util.*

interface PatientService {
    fun create(patientDTO: PatientDTO): PatientDTO
    fun findById(id: UUID): PatientDTO
    fun findAll(): List<PatientDTO>
    fun findByHospitalId(hospitalId: UUID): List<PatientDTO>
    fun findByServiceId(serviceId: UUID): List<PatientDTO>
    fun findBySocialSecurityNumber(socialSecurityNumber: String): PatientDTO
    fun searchByName(name: String): List<PatientDTO>
    fun update(id: UUID, patientDTO: PatientDTO): PatientDTO
    fun deleteById(id: UUID)
}